internal/providers/docker/components/container/build.go

```diff
@@ -91,13 +94,39 @@ func (c *Container) buildImage(ctx context.Context) error {
 		resp, err := c.Docker.ImageBuild(ctx, buildContext, opts)
 		fmt.Println("after build")
 		if resp.Body != nil {
+			defer resp.Body.Close()
 			// TODO [DOCKER_PROGRESS]: Capture progress.
-			_, _ = io.Copy(os.Stdout, resp.Body)
-			_ = resp.Body.Close()
+			//_, _ = io.Copy(os.Stdout, resp.Body)
+			scanner := bufio.NewScanner(resp.Body)
+			for scanner.Scan() {
+				var d struct {
+					ErrorDetail struct {
+						Code    int    `json:"code"`
+						Message string `json:"message"`
+					} `json:"errorDetail"`
+					Aux struct {
+						ID string `json:"ID"`
+					} `json:"aux"`
+				}
+				err := json.Unmarshal(scanner.Bytes(), &d)
+				if err != nil {
+					return err
```

Comment:
Two Go style nitpicks here:

1) Only using `err` in this very next condition, so better to do `if err := ...; err != nil {` idiom.
2) `if err != nil { return err }` should be extremely rare. It's almost always better to add context from the call site. In this specific case, json unmarshaling errors (like most of the stdlib and well behaved code) doesn't mention json on marshalling at all, so you should add context before returning the error. Something like `return fmt.Errorf("unmarshalling response: %w", err)` or something like that is good enough.